File: cdmsbats/src/batcommon/datareader/DataRecord.java
/**
 * 
 */
package batcommon.datareader;

import batcommon.datareader.CDMSRawFileStream;

/**
 * @author tdoughty1
 *
 */
public class DataRecord {

	// Primary Function Call for data record
    // Overload to provide optional argument support 
	
	void ReadRecord(CDMSRawFileStream filePtr, int RecordLength){
	
		int mode = 0;
		boolean debug = false;
		
        int[] Record = filePtr.ReadWords(RecordLength, mode);

        this.StoreValues(Record);

        if(debug) {
            this.PrintValues();
        }
	}
	
	void ReadRecord(CDMSRawFileStream filePtr, int RecordLength, int mode){
		
		boolean debug = false;
		
        int[] Record = filePtr.ReadWords(RecordLength, mode);

        this.StoreValues(Record);

        if(debug) {
            this.PrintValues();
        }
	}

	void ReadRecord(CDMSRawFileStream filePtr, int RecordLength, boolean debug){
        
		int mode = 0;
		
        int[] Record = filePtr.ReadWords(RecordLength, mode);

        this.StoreValues(Record);

        if(debug) {
            this.PrintValues();
        }
		
	}
	
	void ReadRecord(CDMSRawFileStream filePtr, int RecordLength, int mode, boolean debug){
		
        int[] Record = filePtr.ReadWords(RecordLength, mode);

        this.StoreValues(Record);

        if(debug) {
            this.PrintValues();
        }
	}
	
	// Dummy Function - should be overloaded by Inheriting Classes
	void StoreValues(int[] Record){
	}
	
	// Dummy Function - should be overloaded by Inheriting Classes
	void PrintValues(){
	}
}

File: scripting/python/cdmsbats/DataRecord.py
# -*- coding: utf-8 -*-
"""
Created on Fri Apr 11 12:27:34 2014

@author: tdoughty1
"""


class DataRecord:

    # Primary Function Call for data record
    # Responsibility of ReadRecord to return filePtr to starting point
    def ReadRecord(self, filePtr, RecordLength, mode='int32', debug=False):

        Record = filePtr.ReadWords(RecordLength, mode)

        self.StoreValues(Record)

        if debug:
            self.PrintValues()

    # Dummy Function - should be overloaded by Inheriting Classes
    def StoreValues(self, Record):
        pass

    # Dummy Function - should be overloaded by Inheriting Classes
    def PrintValues(self):
        pass
